nifty-ssl/src/main/java/com/facebook/nifty/ssl/TicketSeedFileParser.java

```diff
@@ -0,0 +1,172 @@
+
+    /*
+     * Randomly generated salt to use for the purpose of ticket seeds. The salt in the HKDF is meant to be public.
+     * We do not want to use a random salt so that every machine that gets the same seed will compute the same
+     * ticket keys.
+    */
+    private static final byte[] DEFAULT_TICKET_SALT = CryptoUtil.decodeHex("b78973d13c2d0eb24cf94cd692239867");
+
+    private final byte[] salt;
+
+    /**
+     * Creates a ticket seed file parser with the default salt value.
+     */
+    public TicketSeedFileParser() {
+        this(DEFAULT_TICKET_SALT);
+    }
+
+    /**
+     * Creates a ticket seed file parser with the given binary salt value. All servers in a given logical tier
+     * should use the same salt value (otherwise tickets will not be interchangeable between servers), but
+     * different tiers of machines should use different, unique-per-tier salt values.
+     * The salts do not need to be kept secret.
+     *
+     * @param salt the binary salt value to use. Must not be null.
+     */
+    public TicketSeedFileParser(byte[] salt) {
+        Preconditions.checkNotNull(salt);
+        // Copy the salt array to make sure the one we have is not modified later
+        this.salt = new byte[salt.length];
+        System.arraycopy(salt, 0, this.salt, 0, salt.length);
+    }
+
+    /**
+     * Creates a ticket seed file parser with the given hex-encoded salt value.
+     *
+     * @param hexSalt the hex-encoded salt value to use. Must not be null.
+     */
+    public TicketSeedFileParser(String hexSalt) {
+        this(CryptoUtil.decodeHex(hexSalt));
+    }
+
+    /**
+     * Returns a list of tickets parsed from the ticket file. The keys are returned in a format suitable for use
+     * with netty. The first keys are the current keys, following that are the new keys and old keys.
+     *
+     * @param file the ticket seed file.
+     * @return a list of ticket keys. Current keys are first, then new keys, then old keys.
+     * @throws IOException if reading the file or parsing the JSON fails.
+     * @throws IllegalArgumentException if the JSON file does not contain any current seeds.
+     */
+    public List<SessionTicketKey> parse(final File file) throws IOException, IllegalArgumentException {
+        List<String> allSeeds = TicketSeeds.parseFromJSONFile(file).getAllSeeds();
+        return allSeeds.stream().map(this::deriveKeyFromSeed).collect(Collectors.toList());
+    }
+
+    /**
+     * Helper class used to represent the contents of a parsed ticket seed file.
+     */
+    private static class TicketSeeds {
+        final List<String> currentSeeds;
+        final List<String> newSeeds;
+        final List<String> oldSeeds;
+
+        TicketSeeds(final List<String> currentSeeds, final List<String> newSeeds, final List<String> oldSeeds) {
+            Preconditions.checkNotNull(currentSeeds);
+            Preconditions.checkNotNull(newSeeds);
```

Comment:
These aren't necessary since `ImmutableList.copyOf()` will throw on null. Or add a message indicating which parameter is null.